bentoml/_internal/frameworks/xgboost.py

```diff
@@ -31,123 +31,86 @@
+        booster = bentoml.xgboost.load_model('booster_tree')
+    """  # noqa: LN001
+    if not isinstance(bento_model, bentoml.Model):
+        bento_model = get(bento_model)
+        assert isinstance(bento_model, bentoml.Model)
 
-        # `load` the booster back in memory:
-        booster = bentoml.xgboost.load('booster_tree', booster_params=dict(gpu_id=0))
-
-    """  # noqa
-    _, _model_file, _booster_params = _get_model_info(tag, booster_params, model_store)
+    model_file = bento_model.path_of(MODEL_FILENAME)
+    booster = xgb.core.Booster(model_file=model_file)
+    return booster
 
-    return xgb.core.Booster(
-        params=_booster_params,
-        model_file=_model_file,
-    )
 
-
-def save(
+def save_model(
     name: str,
-    model: "xgb.core.Booster",
+    model: xgb.core.Booster,
     *,
-    booster_params: t.Optional[t.Dict[str, t.Union[str, int]]] = None,
-    labels: t.Optional[t.Dict[str, str]] = None,
-    custom_objects: t.Optional[t.Dict[str, t.Any]] = None,
-    metadata: t.Optional[t.Dict[str, t.Any]] = None,
+    signatures: dict[str, ModelSignatureDict] | None = None,
+    labels: dict[str, str] | None = None,
+    custom_objects: dict[str, t.Any] | None = None,
+    metadata: dict[str, t.Any] | None = None,
 ) -> Tag:
     """
-    Save a model instance to BentoML modelstore.
+    Save an XGBoost model instance to the BentoML model store.
 
     Args:
-        name (:code:`str`):
-            Name for given model instance. This should pass Python identifier check.
-        model (`xgboost.core.Booster`):
-            Instance of model to be saved
-        booster_params (:code:`Dict[str, Union[str, int]]`, `optional`, default to :code:`None`):
-            Params for booster initialization
-        labels (:code:`Dict[str, str]`, `optional`, default to :code:`None`):
-            user-defined labels for managing models, e.g. team=nlp, stage=dev
-        custom_objects (:code:`Dict[str, Any]]`, `optional`, default to :code:`None`):
-            user-defined additional python objects to be saved alongside the model,
-            e.g. a tokenizer instance, preprocessor function, model configuration json
-        metadata (:code:`Dict[str, Any]`, `optional`, default to :code:`None`):
-            Custom metadata for given model.
-        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
-            BentoML modelstore, provided by DI Container.
-
+        name (``str``):
+            The name to give to the model in the BentoML store. This must be a valid
+            :obj:`~bentoml.Tag` name.
+        model (:obj:`~xgboost.core.Booster`):
+            The XGBoost model to be saved.
+        signatures (``dict[str, ModelSignatureDict]``, optional):
+            Signatures of predict methods to be used. If not provided, the signatures default to 
+            ``{"predict": {"batchable": False}}``. See :obj:`~bentoml.types.ModelSignature` for more
```

Comment:
looks like we currently don't expose this type in bentoml.types?

Reply:
I think I fixed it in my other PR. Probably should put this one on top of it.